FubarDev.FtpServer/FileUploadEventArgs.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+
+using JetBrains.Annotations;
+
+namespace FubarDev.FtpServer
+{
+    /// <summary>
+    /// Event arguments for a file upload event
+    /// </summary>
+    public class FileUploadEventArgs : EventArgs
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileUploadEventArgs"/> class.
+        /// </summary>
+        /// <param name="connection">The connection of the event</param>
+        /// <param name="fileName">The filename of the event</param>
+        public FileUploadEventArgs([NotNull] FtpConnection connection, [NotNull] string fileName)
+        {
+            Connection = connection;
+            FileName = fileName;
+        }
+
+        /// <summary>
+        /// Gets the connection for this event
+        /// </summary>
+        [NotNull]
+        public FtpConnection Connection { get; }
```

Comment:
Předávej dál jen to, co z toho FTPConnection potřebujeme - takhle má ten, kdo ten event dostane možnost volat jakoukoliv public metodu toho FTPConnection a zbytečně to zvyšuje provázanost kódu (tight couplig).